### PR TITLE
Remove opset usages for transformations

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Remove opset usages for transformations
 - Use operation versions instead of opsets

### Tickets:
 - CVS-118087

### CODE REVIEW

**Comment by olpipi** [comment]
Second try of https://github.com/openvinotoolkit/openvino/pull/22725

File: src/common/transformations/CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,7 @@ if((CMAKE_COMPILER_IS_GNUCXX OR OV_COMPILER_IS_CLANG) AND CMAKE_CXX_STANDARD GRE
     set(CMAKE_CXX_FLAGS "-Wno-error=deprecated ${CMAKE_CXX_FLAGS}")
 endif()
 
-add_library(${TARGET_NAME}_obj OBJECT ${LIBRARY_SRC} ${PUBLIC_HEADERS})
+add_library(${TARGET_NAME}_obj STATIC ${LIBRARY_SRC} ${PUBLIC_HEADERS})
```
Comment (itikhono) [inline src/common/transformations/CMakeLists.txt:23]: OBJECT is expected here
please delete this change

**Review by itikhono — APPROVED** [review]

File: src/common/transformations/tests/common_optimizations/nop_elimination.cpp [hunk 2]
```diff
@@ -1204,51 +1202,51 @@ TEST_P(SplitConcatElimination, eliminate_split_concat_subgraph) {
...
     } else if (p.rnn_type == RNNType::GRU) {
-        sequence = make_shared<ov::opset9::GRUSequence>(data, H, seq_lengths, W, R, B, hidden_size, direction);
-        data = make_shared<ov::opset9::Squeeze>(sequence->output(0), axis_const);
+        sequence = make_shared<ov::op::v5::GRUSequence>(data, H, seq_lengths, W, R, B, hidden_size, direction);
+        data = make_shared<ov::op::v0::Squeeze>(sequence->output(0), axis_const);
         params = {H, seq_lengths, W, R, B};
     } else if (p.rnn_type == RNNType::LSTM) {
-        sequence = make_shared<ov::opset9::LSTMSequence>(data, H, C, seq_lengths, W, R, B, hidden_size, direction);
-        data = make_shared<ov::opset9::Squeeze>(sequence->output(0), axis_const);
+        sequence = make_shared<ov::op::v0::LSTMSequence>(data, H, C, seq_lengths, W, R, B, hidden_size, direction);
```
Comment (itikhono) [inline src/common/transformations/tests/common_optimizations/nop_elimination.cpp:1227]: should it be v5 LSTM Sequence??

**Review by rkazants — APPROVED** [review]

**Review by t-jankowski — APPROVED** [review]

File: src/common/transformations/CMakeLists.txt (hunk 1 above)
Comment (olpipi) [reply src/common/transformations/CMakeLists.txt]: done

File: src/common/transformations/tests/common_optimizations/nop_elimination.cpp (hunk 2 above)
Comment (olpipi) [reply src/common/transformations/tests/common_optimizations/nop_elimination.cpp]: done

**Pull request merged**